src/engine/guidance/collapse_scenario_detection.cpp

```diff
@@ -123,7 +123,7 @@ bool isStaggeredIntersection(const RouteStepIterator step_prior_to_intersection,
 
     const auto angle = [](const RouteStep &step) {
         const auto &intersection = step.intersections.front();
-        const auto entry_bearing = intersection.bearings[intersection.in];
+        const auto entry_bearing = util::bearing::reverse(intersection.bearings[intersection.in]);
```

Comment:
'in' bearing in the intersection struct is stored opposite to the driving direction. You can ensure about it in the assembleSteps function [here](https://github.com/Project-OSRM/osrm-backend/blob/master/include/engine/guidance/assemble_steps.hpp#L183)

'in' bearing from intersection also reversed [here](https://github.com/Project-OSRM/osrm-backend/blob/master/src/engine/guidance/collapse_turns.cpp#L38) and [here](https://github.com/Project-OSRM/osrm-backend/blob/master/src/engine/guidance/post_processing.cpp#L227) while passing to bearing::angleBetween.

But now I'm confused with the bearings::angleBetween function:
```
bearings::angleBetween(0, 0) == bearings::angleBetween(180, 180) == 180;
bearings::angleBetween(0, 180) == bearings::angleBetween(180, 0) == 0;
```
Seems like it's made specially for the opposite driving direction 'in' bearing. In that case, the logic in collapse_turns.cpp or post_processing.cpp should also be reviewed.